src/test/java/com/commercetools/sync/sdk2/commons/utils/ClientConfigurationUtilsTest.java

```diff
@@ -0,0 +1,25 @@
+package com.commercetools.sync.sdk2.commons.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.commercetools.api.client.ProjectApiRoot;
+import io.vrap.rmf.base.client.oauth2.ClientCredentials;
+import io.vrap.rmf.base.client.oauth2.ClientCredentialsBuilder;
+import org.junit.jupiter.api.Test;
+
+class ClientConfigurationUtilsTest {
+
+  @Test
+  void createClient_WithConfig_ReturnsClient() {
+    final ClientCredentials clientConfig =
+        new ClientCredentialsBuilder()
+            .withClientId("client-id")
+            .withClientSecret("client-secret")
+            .withScopes("scopes")
+            .build();
+    final ProjectApiRoot client =
+        ClientConfigurationUtils.createClient("project-key", clientConfig, "", "");
+
+    assertThat(client.getProjectKey()).isEqualTo("project-key");
```

Comment:
I think we can add one more assertion to verify client config

Reply:
Client config is secret, I don't think it is a good idea to verify it. We need to verify if we are getting client with proper project-key(which is good enough here).